diff --git a/src/apps/experimental/features/preferences/components/DisplayPreferences.tsx b/src/apps/experimental/features/preferences/components/DisplayPreferences.tsx
index 0668ccdbbe88..02e9e428d5de 100644
--- a/src/apps/experimental/features/preferences/components/DisplayPreferences.tsx
+++ b/src/apps/experimental/features/preferences/components/DisplayPreferences.tsx
@@ -226,6 +226,24 @@ export function DisplayPreferences({ onChange, values }: Readonly<DisplayPrefere
                     {globalize.translate('EnableBlurHashHelp')}
                 </FormHelperText>
             </FormControl>
+            <Typography variant='h2'>{globalize.translate('ItemCard')}</Typography>
+
+            <FormControl fullWidth>
+                <FormControlLabel
+                    aria-describedby='display-settings-card-ratings-description'
+                    control={
+                        <Checkbox
+                            checked={values.cardRatings}
+                            onChange={onChange}
+                        />
+                    }
+                    label={globalize.translate('LabelCardRatings')}
+                    name='cardRatings'
+                />
+                <FormHelperText id='display-settings-card-ratings-description'>
+                    {globalize.translate('CardRatingsHelp')}
+                </FormHelperText>
+            </FormControl>
         </Stack>
     );
 }
diff --git a/src/apps/experimental/features/preferences/hooks/useDisplaySettings.ts b/src/apps/experimental/features/preferences/hooks/useDisplaySettings.ts
index 99bd1a812876..15c00e9d192c 100644
--- a/src/apps/experimental/features/preferences/hooks/useDisplaySettings.ts
+++ b/src/apps/experimental/features/preferences/hooks/useDisplaySettings.ts
@@ -85,6 +85,7 @@ async function loadDisplaySettings({
 
     const displaySettings = {
         customCss: settings.customCss() || '',
+        cardRatings: Boolean(settings.cardRatings()),
         dashboardTheme: settings.dashboardTheme() || defaultTheme?.id || FALLBACK_THEME_ID,
         dateTimeLocale: settings.dateTimeLocale() || 'auto',
         disableCustomCss: Boolean(settings.disableCustomCss()),
@@ -132,6 +133,7 @@ async function saveDisplaySettings({
         userSettings.language(normalizeValue(newDisplaySettings.language));
     }
     userSettings.customCss(normalizeValue(newDisplaySettings.customCss));
+    userSettings.cardRatings(newDisplaySettings.cardRatings);
     userSettings.dashboardTheme(newDisplaySettings.dashboardTheme);
     userSettings.dateTimeLocale(normalizeValue(newDisplaySettings.dateTimeLocale));
     userSettings.disableCustomCss(newDisplaySettings.disableCustomCss);
diff --git a/src/apps/experimental/features/preferences/types/displaySettingsValues.ts b/src/apps/experimental/features/preferences/types/displaySettingsValues.ts
index 1f5f9ba15cca..ea480469e7ad 100644
--- a/src/apps/experimental/features/preferences/types/displaySettingsValues.ts
+++ b/src/apps/experimental/features/preferences/types/displaySettingsValues.ts
@@ -1,5 +1,6 @@
 export interface DisplaySettingsValues {
     customCss: string;
+    cardRatings: boolean;
     dashboardTheme: string;
     dateTimeLocale: string;
     disableCustomCss: boolean;
diff --git a/src/components/cardbuilder/card.scss b/src/components/cardbuilder/card.scss
index a013b34d6386..221566aaf765 100644
--- a/src/components/cardbuilder/card.scss
+++ b/src/components/cardbuilder/card.scss
@@ -878,3 +878,52 @@ button::-moz-focus-inner {
     transform: scale(1.4, 1.4);
     transition: 0.2s;
 }
+
+.cardRatingContainer {
+    display: flex;
+    flex-direction: column;
+    align-items: flex-start;
+    position: absolute;
+    top: 3px;
+    left: 3px;
+    padding: 5px;
+    font-weight: 550;
+    background-color: rgba(0, 0, 0, 0.5);
+    border-radius: 15px;
+    z-index: 10;
+}
+
+.card-hoverable:hover .cardRatingContainer {
+    opacity: 1;
+}
+
+.cardCriticRating {
+    background-repeat: no-repeat;
+    background-size: auto 1.2em;
+    min-height: 1.2em;
+    background-position: 2px center;
+    padding-left: 24px;
+}
+
+.cardRatingFresh {
+    background-image: url(../../assets/img/fresh.svg);
+}
+
+.cardRatingRotten {
+    background-image: url(../../assets/img/rotten.svg);
+}
+
+.cardCommunityRating {
+    min-height: 1.2em;
+    display: flex;
+    align-items: center;
+    margin-right: 3px;
+}
+
+.cardStarIcon {
+    width: auto !important;
+    height: auto !important;
+    font-size: 1.4em;
+    margin-right: 3px;
+    color: #f2b01e;
+}
diff --git a/src/components/cardbuilder/cardBuilder.js b/src/components/cardbuilder/cardBuilder.js
index 45c5bba0c091..84154cc941f0 100644
--- a/src/components/cardbuilder/cardBuilder.js
+++ b/src/components/cardbuilder/cardBuilder.js
@@ -42,6 +42,8 @@ import {
     resolveMixedShapeByAspectRatio
 } from './cardBuilderUtils';
 
+import * as userSettings from 'scripts/settings/userSettings';
+
 const enableFocusTransform = !browser.slow && !browser.edge;
 
 /**
@@ -1134,12 +1136,36 @@ function buildCard(index, item, apiClient, options) {
     let additionalCardContent = '';
 
     if (layoutManager.desktop && !options.disableHoverMenu) {
+        additionalCardContent += getRatingHtml(item);
         additionalCardContent += getHoverMenuHtml(item, action);
     }
 
     return '<' + tagName + ' data-index="' + index + '"' + timerAttributes + actionAttribute + ' data-isfolder="' + (item.IsFolder || false) + '" data-serverid="' + (item.ServerId || options.serverId) + '" data-id="' + (item.Id || item.ItemId) + '" data-type="' + item.Type + '"' + mediaTypeData + collectionTypeData + channelIdData + pathData + positionTicksData + collectionIdData + playlistIdData + contextData + parentIdData + startDate + endDate + ' data-prefix="' + escapeHtml(prefix) + '" class="' + className + '"' + ariaLabelAttribute + '>' + cardImageContainerOpen + innerCardFooter + cardImageContainerClose + overlayButtons + additionalCardContent + cardScalableClose + outerCardFooter + cardBoxClose + '</' + tagName + '>';
 }
 
+/**
+ * Generates HTML markup for card rating.
+ * @param {object} item - Item used to generate the card rating.
+ * @returns {string} HTML markup of the card rating.
+ */
+function getRatingHtml(item) {
+    const showRatings = userSettings.cardRatings();
+    if (!showRatings) {
+        return '';
+    }
+
+    let cardRatingHtml = '';
+    if (item.CommunityRating) {
+        cardRatingHtml += `<div class="cardRating"><span class="material-icons star cardRatingIcon cardCommunityRating" aria-hidden="true"></span><span>${item.CommunityRating.toFixed(1)}</span></div>`;
+    }
+    if (item.CriticRating) {
+        const backgroundImageClass = item.CriticRating >= 60 ? 'cardRatingFresh' : 'cardRatingRotten';
+        cardRatingHtml += `<div class="cardRating"><span class="cardRatingIcon cardCriticRating ${backgroundImageClass}"></span><span>${item.CriticRating}</span></div>`;
+    }
+
+    return cardRatingHtml ? `<div class="cardRatingContainer">${cardRatingHtml}</div>` : '';
+}
+
 /**
  * Generates HTML markup for the card overlay.
  * @param {object} item - Item used to generate the card overlay.
diff --git a/src/components/displaySettings/displaySettings.js b/src/components/displaySettings/displaySettings.js
index 72b976c2a5c0..59bc27a0a221 100644
--- a/src/components/displaySettings/displaySettings.js
+++ b/src/components/displaySettings/displaySettings.js
@@ -123,6 +123,7 @@ function loadForm(context, user, userSettings) {
     context.querySelector('#chkFadein').checked = userSettings.enableFastFadein();
     context.querySelector('#chkBlurhash').checked = userSettings.enableBlurhash();
     context.querySelector('#chkBackdrops').checked = userSettings.enableBackdrops();
+    context.querySelector('#selectCardRatings').value = userSettings.cardRatings();
     context.querySelector('#chkDetailsBanner').checked = userSettings.detailsBanner();
 
     context.querySelector('#chkDisableCustomCss').checked = userSettings.disableCustomCss();
@@ -171,6 +172,7 @@ function saveUser(context, user, userSettingsInstance, apiClient) {
     userSettingsInstance.enableFastFadein(context.querySelector('#chkFadein').checked);
     userSettingsInstance.enableBlurhash(context.querySelector('#chkBlurhash').checked);
     userSettingsInstance.enableBackdrops(context.querySelector('#chkBackdrops').checked);
+    userSettingsInstance.cardRatings(context.querySelector('#selectCardRatings').value);
     userSettingsInstance.detailsBanner(context.querySelector('#chkDetailsBanner').checked);
 
     userSettingsInstance.disableCustomCss(context.querySelector('#chkDisableCustomCss').checked);
diff --git a/src/components/displaySettings/displaySettings.template.html b/src/components/displaySettings/displaySettings.template.html
index edaf2662213f..4bf3d8bf376e 100644
--- a/src/components/displaySettings/displaySettings.template.html
+++ b/src/components/displaySettings/displaySettings.template.html
@@ -300,6 +300,18 @@ <h2 class="sectionTitle">
         <div class="fieldDescription checkboxFieldDescription">${UseEpisodeImagesInNextUpHelp}</div>
     </div>
 
+    <h2 class="sectionTitle">
+        ${ItemCard}
+    </h2>
+
+	<div class="checkboxContainer checkboxContainer-withDescription">
+		<label>
+			<input type="checkbox" is="emby-checkbox" id="chkCardRatings" />
+			<span>${ShowRatingsOnCards}</span>
+		</label>
+		<div class="fieldDescription checkboxFieldDescription">${ShowRatingsOnCardsHelp}</div>
+	</div>
+
     <h2 class="sectionTitle">
         ${ItemDetails}
     </h2>
diff --git a/src/scripts/settings/userSettings.js b/src/scripts/settings/userSettings.js
index b10389ccbf3f..1a975dec7237 100644
--- a/src/scripts/settings/userSettings.js
+++ b/src/scripts/settings/userSettings.js
@@ -311,6 +311,19 @@ export class UserSettings {
         return this.get('customCss', false);
     }
 
+    /**
+    * Get or set 'cardRatings' state.
+    * @param {boolean|undefined} [val] - Whether to show ratings on cards.
+    * @return {boolean} Whether ratings are shown on cards.
+    */
+    cardRatings(val) {
+        if (val !== undefined) {
+            return this.set('cardRatings', val.toString(), false);
+        }
+
+        return this.get('cardRatings', 'true') !== 'false';
+    }
+
     /**
      * Get or set 'Details Banner' state.
      * @param {boolean|undefined} [val] - Flag to enable 'Details Banner' or undefined.
@@ -695,6 +708,7 @@ export const enableThemeVideos = currentSettings.enableThemeVideos.bind(currentS
 export const enableFastFadein = currentSettings.enableFastFadein.bind(currentSettings);
 export const enableBlurhash = currentSettings.enableBlurhash.bind(currentSettings);
 export const enableBackdrops = currentSettings.enableBackdrops.bind(currentSettings);
+export const cardRatings = currentSettings.cardRatings.bind(currentSettings);
 export const detailsBanner = currentSettings.detailsBanner.bind(currentSettings);
 export const useEpisodeImagesInNextUpAndResume = currentSettings.useEpisodeImagesInNextUpAndResume.bind(currentSettings);
 export const language = currentSettings.language.bind(currentSettings);
diff --git a/src/strings/en-us.json b/src/strings/en-us.json
index 11911ec3994c..423f25c835a8 100644
--- a/src/strings/en-us.json
+++ b/src/strings/en-us.json
@@ -151,6 +151,7 @@
     "Bwdif": "Bob Weaver DeInterlacing Filter (BWDIF)",
     "CancelRecording": "Cancel recording",
     "CancelSeries": "Cancel series",
+    "CardRatingsHelp": "Allows title cards to display critic/community ratings, when available.",
     "Casual": "Casual",
     "Categories": "Categories",
     "ChangingMetadataImageSettingsNewContent": "Changes to metadata or artwork downloading settings will only apply to new content added to your library. To apply the changes to existing titles, you'll need to refresh their metadata manually.",
@@ -585,6 +586,7 @@
     "Inker": "Inker",
     "InstallingPackage": "Installing {0} (version {1})",
     "InstantMix": "Instant mix",
+    "ItemCard": "Item Card",
     "ItemCount": "{0} items",
     "ItemDetails": "Item Details",
     "Items": "Items",
@@ -653,6 +655,7 @@
     "LabelCachePath": "Cache path",
     "LabelCachePathHelp": "Specify a custom location for server cache files such as images. Leave blank to use the server default.",
     "LabelCancelled": "Cancelled",
+    "LabelCardRatings": "Item Card Ratings Display",
     "LabelCertificatePassword": "Certificate password",
     "LabelCertificatePasswordHelp": "If your certificate requires a password, please enter it here.",
     "LabelChannels": "Channels",
